lib/cfa/augeas_parser/writer.rb

```diff
@@ -0,0 +1,114 @@
+module CFA
+  # Smart writer trying to do as less modification as possible.
+  # @note internal only, unstable API
```

Comment:
yes, it is tested, you can check test coverage and also check new tests that verify it is correctly written.